lib/tests/formulae.rb

```diff
@@ -607,22 +607,30 @@ def formula!(formula_name)
           bottle_reinstall_formula(formula, new_formula)
           test "brew", "linkage", "--test", formula_name
 
-          if formula.test_defined?
+          test_passed_or_missing = if formula.test_defined?
             test "brew", "install", "--only-dependencies", "--include-test",
                                     formula_name
+            # Intentionally not passing --retry here to avoid papering over
+            # flaky tests when a formula isn't being pulled in as a dependent.
             test "brew", "test", "--verbose", formula_name
+
+            steps.last.passed?
```

Comment:
Considering it is called `test_passed_or_missing`, don't we want:

```suggestion
            steps.last.passed?
          else
            true
```

Reply:
Yes, it was there in a previous iteration, I swear 😅.

Reply:
https://github.com/Homebrew/homebrew-test-bot/pull/447